### PR TITLE
flatten 2D or 3D array

### PR DESCRIPTION
**Pull request by dsope05**
recursive solution

### CODE REVIEW

File: solutions/116.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// flatten the array
+//
+const solution = (arr) => {
+  if (arr.length === 0) {
+    return arr;
+  }
+  let a = arr.shift();
```
Comment (yjlim5) [inline solutions/116.js]: variable naming- what does `a` represent?

File: solutions/116.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  if (a.length) {
+  a.forEach((elem) => {
```
Comment (yjlim5) [inline solutions/116.js]: indentation issue

File: solutions/116.js (hunk 1 above)
Comment (dsope05) [inline solutions/116.js]: fixed

File: solutions/116.js (hunk 2 above)
Comment (dsope05) [inline solutions/116.js]: fixed

**Pull request merged**